spring-cloud-aws-autoconfigure/src/main/resources/META-INF/spring.factories

```diff
@@ -12,3 +12,5 @@ io.awspring.cloud.autoconfigure.messaging.SnsAutoConfiguration,\
 io.awspring.cloud.autoconfigure.jdbc.AmazonRdsDatabaseAutoConfiguration,\
 io.awspring.cloud.autoconfigure.metrics.CloudWatchExportAutoConfiguration,\
 io.awspring.cloud.autoconfigure.security.CognitoAuthenticationAutoConfiguration
+org.springframework.cloud.client.discovery.EnableDiscoveryClient=\
+org.springframework.cloud.aws.cloudmap.servicediscovery.CloudMapDiscoveryClient
```

Comment:
You are right, this is not required. So I have removed it.

Reply:
Hi @eddumelendez,
Please find my responses below:
* starter should only contain a pom & auto-configuration is only managed at spring-cloud-aws-autoconfigure module
> I followed `spring-cloud-starter-aws-parameter-store-config` module implementation to begin with and since we have write a discovery and auto-registration client part of `spring-cloud-aws-cloudmap` module I am unable to move the autoconfiguration to `autoconfigure` module.

* cloudmap module should not have dependency on spring-boot. So far I have identified it depends on ConfigurationProperties
> Removed `spring-boot` dependencies from the module